File: tests/src/main/java/com/github/klee0kai/test/qualifiers/di/QTestComponent.java
package com.github.klee0kai.test.qualifiers.di;

import com.github.klee0kai.stone.annotations.component.Component;
import com.github.klee0kai.test.qualifiers.QApp;
import com.github.klee0kai.test.qualifiers.di.qualifiers.ProductType;
import com.github.klee0kai.test.qualifiers.di.qualifiers.Token;
import com.github.klee0kai.test.qualifiers.di.qualifiers.UserId;

@Component(
        qualifiers = {ProductType.class, Token.class, UserId.class}
)
public interface QTestComponent {

    public QInetModule inet();

    public QDataModule data();

    void inject(QApp qApp);

    void inject(QApp qApp, ProductType debug);

    void inject(ProductType demo, UserId demo_user_id, Token demo_token,QApp qApp);

    void inject(QApp qApp, Token release_token, ProductType release);
}

File: test_kotlin/src/main/kotlin/com/github/klee0kai/test/qualifiers/di/QTestComponent.kt
package com.github.klee0kai.test.qualifiers.di

import com.github.klee0kai.stone.annotations.component.Component
import com.github.klee0kai.test.qualifiers.QApp
import com.github.klee0kai.test.qualifiers.di.qualifiers.ProductType
import com.github.klee0kai.test.qualifiers.di.qualifiers.Token
import com.github.klee0kai.test.qualifiers.di.qualifiers.UserId

@Component(qualifiers = [ProductType::class, Token::class, UserId::class])
interface QTestComponent {
    fun inet(): QInetModule
    fun data(): QDataModule
    fun inject(qApp: QApp)
    fun inject(qApp: QApp, debug: ProductType)
    fun inject(demo: ProductType, demoUserId: UserId, demoToke: Token, qApp: QApp)
    fun inject(qApp: QApp?, release_token: Token?, release: ProductType?)
}